reana_db/models.py

```diff
@@ -492,3 +493,78 @@ class AuditLog(Base, Timestamp):
     def __repr__(self):
         """Audit log string representation."""
         return "<AuditLog {} {}>".format(self.id_, self.action)
+
+
+class ResourceType(enum.Enum):
+    """Enumeration of resource types."""
+
+    cpu = 0
+    gpu = 1
+    disk = 2
+
+
+class ResourceUnit(enum.Enum):
+    """Enumeration of resource usage units."""
+
+    bytes_ = 0
+    milliseconds = 1
+
+
+class Resource(Base, Timestamp):
+    """Resource table."""
+
+    __tablename__ = "resource"
+    __table_args__ = {"schema": "__reana"}
+
+    id_ = Column(UUIDType, primary_key=True, default=generate_uuid)
+    name = Column(String(1024))
+    type_ = Column(Enum(ResourceType), nullable=False)
+    unit = Column(Enum(ResourceUnit), nullable=False)
+    title = Column(String(1024))
+
+    def __repr__(self):
+        """Resource string representation."""
+        return "<Resource {} {}>".format(self.id_, self.name)
+
+
+class UserResource(Base, Timestamp):
+    """User Resource table."""
+
+    __tablename__ = "user_resource"
+    __table_args__ = {"schema": "__reana"}
+
+    user_id = Column(
+        UUIDType, ForeignKey("__reana.user_.id_"), primary_key=True, nullable=False
+    )
+    resource_id = Column(
+        UUIDType, ForeignKey("__reana.resource.id_"), primary_key=True, nullable=False
+    )
+    quota_limit = Column(BigInteger())
+    quota_used = Column(BigInteger())  # bigint allows us to go up to to 8192 pebibytes
+
+    def __repr__(self):
+        """User Resource string representation."""
+        return "<User {} consumed {} of {} out of {}>".format(
```

Comment:
I wonder if this is a good representation of this model. I'm very used to see `<ModelName ....`, from a first glance it might be confused with a `<User>` object.

Reply:
Amended to follow `<ModelName ....>`